File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/FormFill.cs
using System;
using System.IO;
using Xfinium.Pdf;
using Xfinium.Pdf.Graphics;
using Xfinium.Pdf.Forms;

namespace Xfinium.Pdf.Samples
{
    /// <summary>
    /// FormFill sample.
    /// </summary>
    public class FormFill
    {
        /// <summary>
        /// Main method for running the sample.
        /// </summary>
        /// <param name="stream"></param>
        public static SampleOutputInfo[] Run(Stream stream)
        {
            PdfFixedDocument document = new PdfFixedDocument(stream);
            (document.Form.Fields["firstname"] as PdfTextBoxField).Text = "John";
            (document.Form.Fields["lastname"] as PdfTextBoxField).Value = "Doe";

            (document.Form.Fields["sex"].Widgets[0] as PdfRadioButtonWidget).Checked = true;

            (document.Form.Fields["firstcar"] as PdfComboBoxField).SelectedIndex = 0;

            (document.Form.Fields["secondcar"] as PdfListBoxField).SelectedIndex = 1;

            (document.Form.Fields["agree"] as PdfCheckBoxField).Checked = true;
            document.Form.FlattenFields();

            SampleOutputInfo[] output = new SampleOutputInfo[] { new SampleOutputInfo(document, "xfinium.pdf.sample.formfill.pdf") };
            return output;
        }
    }
}

File: ION/packages/Nfinium/XFINIUM.PDF 5.4.0 Mobile Professional Edition Commercial Version/Samples/Common/FormFill.vb
Imports System.IO
Imports Xfinium.Pdf
Imports Xfinium.Pdf.Graphics
Imports Xfinium.Pdf.Forms

Namespace Xfinium.Pdf.Samples
	''' <summary>
	''' FormFill sample.
	''' </summary>
	Public Class FormFill
		''' <summary>
		''' Main method for running the sample.
		''' </summary>
		''' <param name="stream"></param>
		Public Shared Function Run(stream As Stream) As SampleOutputInfo()
			Dim document As New PdfFixedDocument(stream)
			TryCast(document.Form.Fields("firstname"), PdfTextBoxField).Text = "John"
			TryCast(document.Form.Fields("lastname"), PdfTextBoxField).Value = "Doe"

			TryCast(document.Form.Fields("sex").Widgets(0), PdfRadioButtonWidget).Checked = True

			TryCast(document.Form.Fields("firstcar"), PdfComboBoxField).SelectedIndex = 0

			TryCast(document.Form.Fields("secondcar"), PdfListBoxField).SelectedIndex = 1

			TryCast(document.Form.Fields("agree"), PdfCheckBoxField).Checked = True

			Dim output As SampleOutputInfo() = New SampleOutputInfo() {New SampleOutputInfo(document, "xfinium.pdf.sample.formfill.pdf")}
			Return output
		End Function
	End Class
End Namespace
